numpy/random/generator.pyx

```diff
@@ -3421,18 +3442,35 @@ cdef class Generator:
         --------
         >>> mean = (1, 2)
         >>> cov = [[1, 0], [0, 1]]
-        >>> x = np.random.default_rng().multivariate_normal(mean, cov, (3, 3))
+        >>> rng = np.random.default_rng()
+        >>> x = rng.multivariate_normal(mean, cov, (3, 3))
         >>> x.shape
         (3, 3, 2)
 
+        We can use a different method other than the default to factorize cov:
+        >>> y = rng.multivariate_normal(mean, cov, (3, 3), method='cholesky')
+        >>> y.shape
+        (3, 3, 2)
+
+        We can also use a precomputed factor matrix of cov to speed up the
+        computation. This is very useful when one needs to generate random
+        values using the sample covariance matrix but different means.
+        >>> s, u = np.linalg.eigh(cov)
+        >>> A = u * np.sqrt(s)  # factor matrix of cov using Eigendecomposition.
+        >>> z = rng.multivariate_normal(mean, A, (3, 3), use_factor=True)
+        >>> z.shape
+        (3, 3, 2)
+
         The following is probably true, given that 0.6 is roughly twice the
         standard deviation:
 
         >>> list((x[0,0,:] - mean) < 0.6)
         [True, True] # random
 
         """
-        from numpy.dual import svd
+        if method not in {'eigh', 'svd', 'cholesky'}:
```

Comment:
Nit - Sets are slow for things this small, use a tuple.

Reply:
I always had the understanding that sets were faster for membership tests so this is a common idiom I also use so I put it to the test. At least on my machine there's at most a few ns in it:
```python
In [1]: import psutil

In [2]: p = psutil.Process()

In [3]: p.nice(psutil.REALTIME_PRIORITY_CLASS)

In [4]: p.nice()
Out[4]: <Priority.REALTIME_PRIORITY_CLASS: 256>

In [5]: method = 'eigh'

In [6]: %timeit res = method not in {'eigh', 'svd', 'cholesky'}
122 ns ± 2.86 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [7]: %timeit res = method not in {'eigh', 'svd', 'cholesky'}
118 ns ± 0.899 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [8]: %timeit res = method not in ('eigh', 'svd', 'cholesky')
119 ns ± 2.87 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [9]: %timeit res = method not in ('eigh', 'svd', 'cholesky')
117 ns ± 2.18 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

```

Reply:
Interesting... I assumed that a linear search would be nicer than a hash map for such a small membership size, even if sets are O(1) and tuples are O(n) for membership tests.

Reply:
timing is probably dominated by object creation and teardown

Reply:
Resolving, I don't see any reason to change this to a tuple